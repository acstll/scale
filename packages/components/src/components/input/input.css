/**
 * @prop --input-border: Border of the input
 * @prop --input-border-radius: Border radius of the input
 * @prop --input-padding: Padding of the input
 * @prop --input-font-size: Font size of the input
 * @prop --input-font-family: Font family of the input
 */

.input {
  display: flex;
  width: 100%;
  border: var(--input-border, 1px solid #dfdfdf);
  border-radius: var(--input-border-radius, 4px);
  padding: var(--input-padding, 0.5rem);
  font-size: var(--input-font-size, 16px);
  font-family: var(--input-font-family, inherit);
  box-sizing: border-box;
}

/*
 * TODO: implement accessability
 * -----------------------------
 * Once the user starts tabbing it probably
 * means that a screen-reader is required
 */

.input:not(.tabbing):focus {
  outline: 0;
}
