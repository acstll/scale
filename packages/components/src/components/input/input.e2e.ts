import { newE2EPage } from '@stencil/core/testing';

describe('t-input', () => {
  it('renders', async () => {
    const page = await newE2EPage();
    await page.setContent('<t-input/>');
    const element = await page.find('t-input');
    expect(element).toHaveClass('hydrated');
  });
});
