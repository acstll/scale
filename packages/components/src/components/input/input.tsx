import { Component, Prop, Event, h, EventEmitter } from '@stencil/core';
import { CssClassMap } from '../../utils/utils';
import classNames from 'classnames';

export type InputTypes =
  | 'email'
  | 'hidden'
  | 'number'
  | 'password'
  | 'tel'
  | 'text'
  | 'url';

@Component({
  tag: 't-input',
  styleUrl: 'input.css',
  shadow: true,
})
export class Input {
  /** (optional) Input text class */
  @Prop() public customClass?: string = '';
  /** (optional) Input text theme */
  @Prop() public theme?: string = '';
  /** (optional) Input type */
  @Prop() public type?: InputTypes = 'text';
  /** (optional) Input name */
  @Prop() public name?: string = '';
  /** (optional) Input text value */
  @Prop({ mutable: true }) public value: string;
  /** (optional) Input text error message */
  @Prop({ mutable: true }) public errorMessage: string;
  /** (optional) Input text event changed */
  @Event() public changed: EventEmitter<string>;

  public handleChange(event) {
    this.value = event.target ? event.target.value : this.value;
    this.changed.emit(this.value);
  }

  public render() {
    return (
      <input
        type={this.type}
        class={this.getCssClassMap()}
        value={this.value}
        name={this.name}
        onInput={event => this.handleChange(event)}
      />
    );
  }

  private getCssClassMap(): CssClassMap {
    return classNames(
      'input',
      this.customClass && this.customClass,
      this.theme && `input--theme-${this.theme}`,
      this.type && `input--type-${this.type}`
    );
  }
}
