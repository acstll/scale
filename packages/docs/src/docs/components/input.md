---
title: Input
description: Input element
url: /docs/input
contributors:
  - SylviaGonschior
---

# Input

Input component

selector: `t-input`

## Default

```javascript
<t-input></t-input>
```

<div class="demo-container">
  <t-input></t-input>
</div>
