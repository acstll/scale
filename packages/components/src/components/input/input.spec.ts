import { newSpecPage } from '@stencil/core/testing';
import { Input } from './input';

describe('Input', () => {
  let element;
  beforeEach(async () => {
    element = new Input();
  });

  it('should match snapshot', async () => {
    const page = await newSpecPage({
      components: [Input],
      html: `<t-input></t-input>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should handle a custom css class', () => {
    element.customClass = 'custom';
    expect(element.getCssClassMap()).toContain('custom');
  });

  it('should handle theme css class', () => {
    element.theme = 'default';
    expect(element.getCssClassMap()).toContain('input--theme-default');
  });

  it('should have a default css class', () => {
    expect(element.getCssClassMap()).toContain('input');
  });
});
