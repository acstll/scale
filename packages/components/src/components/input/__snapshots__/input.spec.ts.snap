// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should match snapshot 1`] = `
<t-input>
  <mock:shadow-root>
    <input class="input input--type-text" type="text">
  </mock:shadow-root>
</t-input>
`;
